### PR TITLE
Add test chacha20

### PR DESCRIPTION
**Pull request by vonasek**

### CODE REVIEW

File: neqo-client/src/main.rs [hunk 1]
```diff
@@ -519,6 +526,7 @@ fn main() -> Res<()> {
                     remote_addr,
                     &format!("{}", host),
                     &[url],
+                    ciphers,
```
Comment (martinthomson) [inline neqo-client/src/main.rs:529]: The API on this is starting to get uncomfortably wide.  I wonder if we need to start breaking this down more.

File: neqo-transport/src/connection.rs [hunk 2]
```diff
@@ -511,6 +512,20 @@ impl Connection {
         Ok(())
     }
 
+    /// Enable a set of ciphers.
+    pub fn enable_ciphers(&mut self, ciphers: &[Cipher]) -> Res<()> {
+        if self.role() != Role::Client {
```
Comment (martinthomson) [inline neqo-transport/src/connection.rs:517]: Please remove this check.  The server has the same needs.

File: neqo-transport/src/connection.rs [hunk 3]
```diff
@@ -511,6 +512,20 @@ impl Connection {
...
+            qerror!([self], "Cannot enable ciphers in role {:?}", self.role);
+            return Err(Error::ConnectionState);
+        }
+        if self.state != State::Init {
```
Comment (martinthomson) [inline neqo-transport/src/connection.rs:521]: Can you also add a similar check in SecretAgent::enable_ciphers?  Belt and braces.

File: neqo-client/src/main.rs (hunk 1 above)
Comment (vonasek) [reply neqo-client/src/main.rs:574]: Maybe. For now I've added allow(clippy::too_many_arguments) to make clippy happy.

**Pull request merged**